expression/builtin_arithmetic.go

```diff
@@ -334,6 +340,107 @@ func (s *builtinArithmeticMinusIntSig) evalInt(row []types.Datum) (val int64, is
 	return a - b, false, nil
 }
 
+type arithmeticDivideFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *arithmeticDivideFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	tpA, tpB := args[0].GetType(), args[1].GetType()
+	tcA, tcB := numericContextResultType(tpA), numericContextResultType(tpB)
+	if tcA == types.ClassReal || tcB == types.ClassReal {
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpReal, tpReal, tpReal)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		setFlenDecimal4RealOrDecimal(bf.tp, tpA, tpB, true)
+		sig := &builtinArithmeticDivideRealSig{baseRealBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	} else if tcA == types.ClassDecimal || tcB == types.ClassDecimal {
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDecimal, tpDecimal, tpDecimal)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		setFlenDecimal4RealOrDecimal(bf.tp, tpA, tpB, false)
+		sig := &builtinArithmeticDivideDecimalSig{baseDecimalBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	} else {
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDecimal, tpInt, tpInt)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		setFlenDecimal4Int(bf.tp, tpA, tpB)
+		sig := &builtinArithmeticDivideIntSig{baseDecimalBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	}
+}
+
+type builtinArithmeticDivideRealSig struct{ baseRealBuiltinFunc }
+type builtinArithmeticDivideDecimalSig struct{ baseDecimalBuiltinFunc }
+type builtinArithmeticDivideIntSig struct{ baseDecimalBuiltinFunc }
+
+func (s *builtinArithmeticDivideRealSig) evalReal(row []types.Datum) (float64, bool, error) {
+	sc := s.ctx.GetSessionVars().StmtCtx
+	a, isNull, err := s.args[0].EvalReal(row, sc)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+	b, isNull, err := s.args[1].EvalReal(row, sc)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+	if math.Abs(b) < floatEpsilon {
+		return 0, true, nil
+	}
+
+	return a / b, false, nil
```

Comment:
check overflow:
mysql> select 1e300/1e-20;
ERROR 1690 (22003): DOUBLE value is out of range in '(1e300 / 1e-20)'